Sources/SymbolKit/SymbolGraph/SemanticVersion/SemanticVersion+Comparable.swift

```diff
@@ -0,0 +1,38 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+extension SymbolGraph.SemanticVersion: Comparable {
+    // Although `Comparable` inherits from `Equatable`, it does not provide a new default implementation of `==`, but instead uses `Equatable`'s default synthesised implementation. The compiler-synthesised `==`` is composed of [member-wise comparisons](https://github.com/apple/swift-evolution/blob/main/proposals/0185-synthesize-equatable-hashable.md#implementation-details), which leads to a false `false` when 2 semantic versions differ by only their build metadata identifiers, contradicting SemVer 2.0.0's [comparison rules](https://semver.org/#spec-item-10).
```

Comment:
What are some possible misuses you have in mind? As far as I can tell this is exactly what `==` is designed to do.

Reply:
If a developer thinks that two versions with different build metadata should be considered different but they are considered equal and that developer writes code based on that assumption that has bugs because the assumption is wrong I would consider that an unintended misuse of this equality behavior.

Reply:
I see. Maybe a note in the documentation could help, maybe similar to how `FloatingPoint` does it?

Reply:
`FloatingPoint` is a great example in that it's another well specified comparison behavior that has a few unintuitive cases. I'm sure many developers, myself included, would get some of the NaN behaviors wrong without looking them up. Especially the collection behaviors around sorting and filtering.

In the `FloatingPoint` case it's clearly the right choice to use operators for equality and comparison but even then there have been [discussions around some undesirable behaviors in the specification](https://forums.swift.org/t/comparable-and-floatingpoint-types/16886).

Some of those considerations apply here as well. For example:

```swift
let versions  = try! ["1.2.3-alpha+002", "1.2.3-alpha+001"].map { SemanticVersion(...) }

versions.sorted() // order didn't change because build metadata isn't considered in comparison

versions.contains("1.2.3-alpha+003") // true because build metadata isn't considered in equality check
```

Documentation helps but I'm still thinking about if this would be better addressed at the API level by not using Equatable/Comparable for these behaviors. See [this comment]. In the FloatingPoint case it's clearly worth it but I feel that the semantic versions aren't as frequently compared as floating point numbers are and that semantic versions with build metadata would be more common than not-a-number floating point values. So that's why I'm a bit hesitant on this.